### PR TITLE
add polygon action store api

### PR DESCRIPTION
**Pull request by esoliani**

### CODE REVIEW

File: app/scripts/actions/polygon.es6 [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  loadRegions: {},
+  loadRegionsCompleted: {},
+  loadDistricts: {},
+  loadDistrictsCompleted: {},
+});
+
+// SIDE-EFFECT: xhr request is triggered on polygonActions.loadRegions()
+polygonActions.loadRegions.listen(() => {
+  getRegions()
+    .then(polygonActions.loadRegionsCompleted);
```
Comment (uniphil) [inline app/scripts/actions/polygon.es6]: Add a `.catch` for `loadRegionsFailed`

File: app/scripts/api.es6 [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import ckan from './utils/ckan';
 
-
 const API_ROOT = '//data.takwimu.org/api';
 const resourceUrl = id => `${API_ROOT}/action/datastore_search?resource_id=${id}`;
 // const queryUrl = q => `${API_ROOT}/action/datastore_search_sql?sql=${q}`;
+const staticResource = q => `layers/${q}.json`;
 
+const get = url =>
+  fetch(url).then(resp => resp.json());
```
Comment (uniphil) [inline app/scripts/api.es6:9]: :+1: we can maybe add move [`makeHTTPErrorNice`](https://github.com/WorldBank-Transport/waterdash/blob/waterpoints-loading/app/scripts/utils/ckan.es6#L18) and [`rejectIfNotHTTPOk`](https://github.com/WorldBank-Transport/waterdash/blob/waterpoints-loading/app/scripts/utils/ckan.es6#L30) somewhere to use it here also, after #8 is merged

File: app/scripts/stores/polygon.es6 [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { createStore } from 'reflux';
+import SaneStore from '../utils/sane-store-mixin';
+import { loadRegionsCompleted, loadDistrictsCompleted } from '../actions/polygon';
+
+const PolygonsStore = createStore({
+  initialData: [],
```
Comment (uniphil) [inline app/scripts/stores/polygon.es6]: `initialData` is an array but `setData` makes it an object -- this should probably be `{}`.

**Comment by uniphil** [comment]
@esoliani what would you think about making everything about `polygons` generic, and having `Regions` and `Districts` in an enum to match where we need it?

The actions could be
`loadPolygons()`, and we could use it with `loadPolygons(poly.Districts())` as a payload.

Then the store could either hold `{ type: PolyEnumOption, data: ... }`, or it could just have `data` and we could have another `PolyTypeStore`.

Or, the store could hold a `PolyDataEnum` like this:

``` js
const $ = {};  // placeholder
const PolyData = Enum({
  Regions: $,
  Districts: $,
}, {
  unwrap: () => this.match({
    _: (data) => data,
  }),
});
```

With this, the store can do `setData(PolyData.Regions(data));`, and then components using the data can:
- match on the type, `this.props.data.match({ Regions: () => thing 1, Districts: () => thing 2 })`
- get the data with `unwrap`: `const data = this.props.data.unwrap();`

maybe that's over-complicating things...

**Comment by uniphil** [comment]
Ignore my late-friday complicated comments :smile:

`.catch` on the fetch, `initialData` as the right type are the only two real blockers on this. We can refactor polygon regions/districts stuff later if we want.

File: app/scripts/actions/polygon.es6 (hunk 1 above)
Comment (esoliani) [inline app/scripts/actions/polygon.es6]: fixed

**Comment by uniphil** [comment]
merging and I'll rebase waterpoints-loading on top. should be easy.

**Pull request merged**